src/Workspaces/Core/MSBuild/MSBuild/ProjectFile/Extensions.cs

```diff
@@ -26,7 +26,7 @@ public static IEnumerable<MSB.Framework.ITaskItem> GetMetadataReferences(this MS
         public static IEnumerable<ProjectFileReference> GetProjectReferences(this MSB.Execution.ProjectInstance executedProject)
             => executedProject
                 .GetItems(ItemNames.ProjectReference)
-                .Where(i => !i.HasReferenceOutputAssemblyMetadataEqualToTrue())
+                .Where(i => i.ReferenceOutputAssemblyIsTrue())
```

Comment:
This switch seems strange -- was the original logic with the ! ever right? I see the method's handling of the default case is changing, but the explicitly true/false cases didn't change?

Reply:
Nope. It wasn't ever right and I think it makes more sense now. Essentially, this means that we won't create a project reference if the item has `<ReferenceOutputAssembly>false</ReferenceOutputAssembly>` metadata. However, in any other case (e.g. no metadata at all, metadata has invalid value), we should create a project reference.

When I added the fallback to try and load project references when there isn't an output path to match with it, a test started failing because this logic was incorrect.